airbyte-integrations/connectors/destination-s3/src/main/java/io/airbyte/integrations/destination/s3/template/S3FilenameTemplateManager.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.destination.s3.template;
+
+import static java.util.Optional.ofNullable;
+import static org.apache.commons.lang3.StringUtils.EMPTY;
+
+import io.airbyte.integrations.destination.s3.S3DestinationConstants;
+import java.io.IOException;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.TimeZone;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.apache.commons.text.StringSubstitutor;
+import org.apache.commons.text.lookup.StringLookupFactory;
+
+/**
+ * This class is responsible for building the filename template based on user input, see
+ * file_name_pattern in the specification of connector currently supported only S3 staging.
+ */
+public class S3FilenameTemplateManager {
+
+  private static final String UTC = "UTC";
+  private final StringSubstitutor stringSubstitutor;
+
+  public S3FilenameTemplateManager() {
+    stringSubstitutor = new StringSubstitutor();
+  }
+
+  public String adaptFilenameAccordingSpecificationPatternWithDefaultConfig(final S3FilenameTemplateParameterObject parameterObject)
```

Comment:
I appreciate propensity towards descriptiveness, but this is too much :) let's keep method names under 50 characters when possible. E.g., this could be renamed to `applyPatternToFilename` or something like that.

Reply:
> I appreciate propensity towards descriptiveness, but this is too much :) let's keep method names under 50 characters when possible. E.g., this could be renamed to `applyPatternToFilename` or something like that.

Fixed